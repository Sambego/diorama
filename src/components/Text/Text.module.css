.text {
  margin: 0;
  max-width: 80vw;
  font-size: 3.5vw;
  line-height: 1.7;
}

@media (min-width: 600px) {
  .text {
    font-size: 2.5vw;
  }
}
