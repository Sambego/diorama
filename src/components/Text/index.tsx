import React from "react";
import PropTypes from "prop-types";

import styles from "./Text.module.css";

type TextProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
}>;

function Text({ children, style, className }: TextProps) {
	return (
		<p style={style} className={`${styles.text} diorama-text ${className}`}>
			{children}
		</p>
	);
}

Text.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	style: PropTypes.shape({}),
};

Text.defaultProps = {
	className: "",
	style: {},
};

export default Text;
