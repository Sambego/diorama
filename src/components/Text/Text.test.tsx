import React from 'react';
import renderer from 'react-test-renderer';
import Text from ".";

it('renders correctly', () => {
  const tree = renderer.create(<Text>Text content</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
