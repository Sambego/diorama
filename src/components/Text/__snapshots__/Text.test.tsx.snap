// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<p
  className="_text_1kphb_1 diorama-text "
  style={{}}
>
  Text content
</p>
`;
